packages/disputer/src/disputer.js

```diff
@@ -175,65 +175,51 @@ class Disputer {
       // Create the transaction.
       const dispute = this.financialContract.methods.dispute(disputeableLiquidation.id, disputeableLiquidation.sponsor);
 
-      // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
-      let totalPaid, gasEstimation;
-      try {
-        [totalPaid, gasEstimation] = await Promise.all([
-          dispute.call({ from: this.account }),
-          dispute.estimateGas({ from: this.account })
-        ]);
-      } catch (error) {
-        this.logger.error({
-          at: "Disputer",
-          message: "Cannot dispute liquidation: not enough collateral (or large enough approval) to initiate dispute✋",
-          disputer: this.account,
-          sponsor: disputeableLiquidation.sponsor,
-          liquidation: disputeableLiquidation,
-          totalPaid,
-          error
-        });
-        continue;
-      }
-
-      const txnConfig = {
-        from: this.account,
-        gas: Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit),
-        gasPrice: this.gasEstimator.getCurrentFastPrice()
-      };
-
       this.logger.debug({
         at: "Disputer",
         message: "Disputing liquidation",
-        liquidation: disputeableLiquidation,
-        txnConfig
+        liquidation: disputeableLiquidation
       });
-
-      // Send the transaction or report failure.
-      let receipt;
       try {
-        receipt = await dispute.send(txnConfig);
+        // Get successful transaction receipt and return value or error.
+        const transactionResult = await runTransaction({
+          transaction: dispute,
+          config: {
+            gasPrice: this.gasEstimator.getCurrentFastPrice(),
+            from: this.account,
+            nonce: await this.web3.eth.getTransactionCount(this.account)
```

Comment:
why are we explicitly providing the nonce here? Isn't this generally injected automatically or is this required by the ynatm?

Reply:
The nonce is passed through from the config passed into `ynatm` so I'm pretty sure `ynatm` ONLY works if you hardcode the `nonce` to prevent it from auto increasing between retries

Reply:
This is how the tests in `ynatm` use `nonce` + our liquidator implementation